socorro/unittest/signature/test_rules.py

```diff
@@ -1291,20 +1291,46 @@ def test_predicate_no_match(self):
         assert predicate_result is False
 
     def test_predicate(self):
-        rule = SigTrim()
+        rule = SigFixWhitespace()
         processed_crash = {
             'signature': 'fooo::baar'
         }
         predicate_result = rule.predicate({}, processed_crash)
         assert predicate_result is True
 
     @pytest.mark.parametrize('signature, expected', [
-        ('all   good', 'all   good'),
-        ('all   good     ', 'all   good'),
-        ('    all   good  ', 'all   good'),
+        ('all   good', 'all good'),
+        ('all   good     ', 'all good'),
+        ('    all   good  ', 'all good'),
     ])
-    def test_action_success(self, signature, expected):
-        rule = SigTrim()
+    def test_leading_trailing_whitespace(self, signature, expected):
+        rule = SigFixWhitespace()
+        processed_crash = {
+            'signature': signature
+        }
+        action_result = rule.action({}, processed_crash, [])
+        assert action_result is True
+        assert processed_crash['signature'] == expected
+
+    @pytest.mark.parametrize('signature, expected', [
+        ('all\tgood', 'all good'),
+        ('all\n\ngood', 'all good'),
+    ])
+    def test_non_space_whitespace(self, signature, expected):
+        rule = SigFixWhitespace()
+        processed_crash = {
+            'signature': signature
+        }
+        action_result = rule.action({}, processed_crash, [])
+        assert action_result is True
+        assert processed_crash['signature'] == expected
+
+    @pytest.mark.parametrize('signature, expected', [
+        ('all   good', 'all good'),
+        ('all  |  good', 'all | good'),
+    ])
+    def test_consecutive_spaces(self, signature, expected):
```

Comment:
Is there any reason not to collapse all these test cases into a single one with all the parameterizations? You could use comments and whitespace to annotate what the cases are covering.

Reply:
That makes a lot of sense. I'll fix that now.